cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -134,195 +160,105 @@ func newEtcdClient() (client *etcd.Client) {
 		}
 		return true, nil
 	}); err != nil {
-		log.Fatalf("Timed out after %s waiting for at least 1 synchronized etcd server in the cluster", timeout)
+		return nil, fmt.Errorf("Timed out after %s waiting for at least 1 synchronized etcd server in the cluster. Error: %v", timeout, err)
 	}
-	return client
+	return client, nil
+}
+
+func getKubeMasterUrl() (string, error) {
+	if *argKubeMasterUrl == "" {
+		return "", fmt.Errorf("no --kube_master_url specified")
+	}
+	parsedUrl, err := url.Parse(os.ExpandEnv(*argKubeMasterUrl))
+	if err != nil {
+		return "", fmt.Errorf("failed to parse --kube_master_url %s - %v", *argKubeMasterUrl, err)
+	}
+	if parsedUrl.Scheme == "" || parsedUrl.Host == "" || parsedUrl.Host == ":" {
```

Comment:
I get that, it just feels like ":" is a pretty random thing to test against - is this some idiom I don't know?